File: Python/BOJ/2606.py
import sys
from collections import deque
# dfs
# def solution():
#     n = int(sys.stdin.readline())
#     pair_len = int(sys.stdin.readline())
#     infected = []
#     stack = [1]
#     graph =[[] for i in range(n+1)]
#     for i in range(pair_len):
#         left, right = map(int, sys.stdin.readline().split())
#         graph[left].append(right)
#         graph[right].append(left)
#     while stack:
#         i = stack.pop()
#         if i not in infected:
#             infected.append(i)
#             for j in graph[i]:
#                 stack.append(j)
#
#     print(len(infected)-1)
#bfs

def bfs(start,graph,infected):
    queue = deque()
    queue.append(start)

    while queue:
        i = queue.popleft()
        for j in graph[i]:
            if j not in infected:
                queue.append(j)
                infected.append(j)


def solution():
    n = int(sys.stdin.readline())
    pair_len = int(sys.stdin.readline())
    infected = []
    graph =[[] for i in range(n+1)]
    for i in range(pair_len):
            left, right = map(int, sys.stdin.readline().split())
            graph[left].append(right)
            graph[right].append(left)
    bfs(1, graph, infected)

    #print(graph)
    print(len(infected) - 1)

solution()

File: C++/BOJ/2606.cpp
#include <iostream>
#include <vector>
#include <stack>
#include <deque>
using namespace std;

void dfs(vector<vector<int>> graph, vector<int> infected)
{
    stack<int> stack;
    stack.push(1);

    while (stack.size() >= 1)
    {
        int i = stack.top();
        stack.pop();
        if (find(infected.begin(), infected.end(),i) == infected.end())
        {
            infected.push_back(i);
            for (auto it : graph[i])
                stack.push(it);
        }
    }
    cout<<infected.size() - 1;
}

void    bfs(vector<vector<int>> graph, vector<int> infected)
{
    deque<int>  queue;

    queue.push_back(1);
    while (queue.size() >= 1)
    {
        int i = queue.front();
        queue.pop_front();
        if (find(infected.begin(), infected.end(), i) == infected.end())
        {
            infected.push_back(i);
            for(auto it :graph[i])
                queue.push_back(it);
        }
    }
    cout<<infected.size() - 1;
}

int main()
{
    int computers;
    cin>>computers;
    int pair_len;
    cin>> pair_len;
    int left, right;
    vector<vector<int>> graph;
    vector<int> infected;
    for(int i = 0; i < computers + 1; i++)
        graph.push_back({});
    for (int i = 0; i < pair_len; i++)
    {
        cin>>left>>right;
        graph[left].push_back(right);
        graph[right].push_back(left);
    }
    dfs(graph, infected);
    bfs(graph, infected);
}